components/producers/producer.go

```diff
@@ -171,12 +174,66 @@ func GetPURLTarget(purlType string, namespace string, name string, version strin
 	return packageurl.NewPackageURL(purlType, namespace, name, version, qualifiers, subpath).ToString()
 }
 
-// EnsureValidPURLTarget takes a purl target string from an untrusted source, e.g. a tool output, and
-// ensures it is a valid purl target according to the packageurl-go library.
+// EnsureValidPURLTarget takes a purl target string from an untrusted source,
+// e.g. a tool output, and ensures it is a valid purl target
 func EnsureValidPURLTarget(purlTarget string) (string, error) {
 	instance, err := packageurl.FromString(purlTarget)
 	if err != nil {
 		return "", err
 	}
 	return instance.ToString(), nil
 }
+
+// GetFileTarget returns a file target string for a given file path.
+// This should be used as the `Issue.Target` field of SAST producers.
+// The root of the `filePath` should be the root of the scanned code.
+//
+// Example: GetFileTarget("src/main.go", 10, 20)
+// Result: "file:///src/main.go:10-20"
+func GetFileTarget(filePath string, startLine int, endLine int) string {
+	if filePath == "" {
+		return ""
+	}
+	url := url.URL{Scheme: "file", Path: filePath}
+
+	return fmt.Sprintf("%s:%d-%d", url.String(), startLine, endLine)
+}
+
+// EnsureValidFileTarget takes a file target string from an untrusted source,
+// e.g. a tool output, and ensures it is a valid file target.
+// file:///path/to/file.txt:10-20
+func EnsureValidFileTarget(fileTarget string) (string, error) {
+	pattern, err := regexp.Compile(`^(.*?:.*?):(.*)$`)
```

Comment:
Thanks, I changed it to a `MustCompile` instead.

I think the current approach is maybe a bit more readable, and because we return the url.URL we get a couple more utility methods.